### DISCUSSION TITLE
Parsing raw c# code (text files)

### DISCUSSION THREAD

**Discussion by mykolamor** [icsharpcode/ILSpy#2159] [Help] [answered]
Hi everyone! I've been a happy user of the ILSpy CSharpDecompiler, specifically the decompiled node tree. But I've been wondering, is there any method to transform raw c# code (text files) into the SyntaxTree node tree? I understand I can create a parser myself, but I'm curious if it already exists.

**Comment by dgrunwald** [comment, marked as answer]
The decompiler's syntax tree and type system are derived from [NRefactory](https://github.com/icsharpcode/NRefactory/) (which had a C# 5 parser), but all portions unneeded for decompilation have been removed to simplify the maintenance.

I recommend using Microsoft.CodeAnalysis.CSharp for parsing C# code.
